src/signing_key.rs

```diff
@@ -173,6 +173,6 @@ impl SigningKey {
 
         let s_bytes = (r + k * self.s).to_bytes();
 
-        Signature { R_bytes, s_bytes }
+        Signature::from_components(R_bytes, s_bytes).unwrap()
```

Comment:
TODO: remove this `unwrap()` by using `signature::Signature::try_sign`, or making `ed25519::Signature::from_components` infallible.